Chapter 3 and beginning of chapter 4.

SirVsbi opened the pull request:
The code so far contains:

-  all the JPA entities and flywal migration files. 
- basic GET AND POST endpoints for products and suppliers (so, no /{id} endpoints yet) 
- repo, service, dto, converter, controller for products & suppliers

Commit:
Initial commit. Add h2. Disable spring security checks. Add springdoc api.

Commit:
Add initial flyway migration files.

Commit:
Add initial Entity files, repo, service and REST controller

Commit:
Add extra entities. Initial endpoints for supply, category and product.

Commit:
Add all the remaining entities. GET product endpoint now contains supplier information.

cristinacalbaza commented on src/main/java/ro/msg/learning/shop/controller/ProductController.java:
```diff
@@ -0,0 +1,45 @@
+    @PostMapping(value = "")
+    public MessageResponse addProduct(@RequestBody ProductDto productDto){
+        try{
+            System.out.println(productDto);
+            System.out.println(productConverter.dtoToModel(productDto));
```
Remove these 2 lines.

cristinacalbaza commented on src/main/java/ro/msg/learning/shop/model/ProductCategory.java:
```diff
@@ -0,0 +1,33 @@
+
+    private String description;
+
+    @OneToMany(mappedBy = "productCategoryId", cascade = CascadeType.ALL, targetEntity = Product.class)
+    private List<Product> products;
```
As we discussed today, please remove the list from here.

cristinacalbaza commented on src/main/java/ro/msg/learning/shop/controller/ProductCategoryController.java:
```diff
@@ -0,0 +1,41 @@
+    @GetMapping(value = "")
+    public ProductCategoryResponse getAllProductCategories(){
+        return new ProductCategoryResponse(
+                (ArrayList<ProductCategoryDto>) productCategoryConverter.convertModelToDto(productCategoryService.getAllProductCategories())
```
You should not access the Data Model from Controller. So, as it is already suggested in Training material, your service methods should receive as parameters a DTO object and return also a DTO. Please change  this for all Services in the app.

Commit:
Implement complete rest API for suppliers. Move converter calls to service.

SirVsbi commented:
Fixed the issues mentioned above. 
Moved the converter calls into the service, now the REST controller only uses DTO's. 
Implemented the rest of the REST API only for the suppliers.
